clients/linux/src/app.rs

```diff
@@ -926,6 +931,153 @@ impl LbApp {
         Ok(())
     }
 
+    fn show_dialog_import_file(&self, parent: Uuid, uris: Vec<String>) -> LbResult<()> {
+        let (d, disk_lbl, lb_lbl, prog_lbl, pbar) = self.gui.new_import_export_dialog(true);
+
+        const FILE_SCHEME: &str = "file://";
+
+        let mut total = 0;
+        let progress = Rc::new(RefCell::new(-1));
+
+        for uri in &uris {
+            if let Some(path) = uri.strip_prefix(FILE_SCHEME) {
+                total += io::get_children_count(Path::new(&path).to_path_buf())?;
+            } else {
+                return Err(LbError::new_user_err(
+                    "Unsupported uri is used.".to_string(),
+                    LbErrTarget::Dialog,
+                ));
+            }
+        }
+
+        d.show_all();
+
+        let ch = make_glib_chan!(self.messenger as m, progress => move |maybe_info: Option<ImportExportFileInfo>| {
+            *progress.borrow_mut() += 1;
+            pbar.set_fraction(*progress.borrow() as f64 / total as f64);
+
+            match maybe_info {
+                None => {
+                    d.close();
+                    m.send(Msg::RefreshTree);
+                }
+                Some(info) => {
+                    lb_lbl.set_text(&info.lockbook_path);
+                    disk_lbl.set_text(&format!("{}", info.disk_path.display()));
+                    prog_lbl.set_text(&format!("{}/{}", *progress.borrow(), total));
+                }
+            }
+            glib::Continue(true)
+        });
+
+        let f = closure!(ch => move |progress: ImportExportFileInfo| {
+            ch.send(Some(progress)).unwrap();
+        });
+
+        spawn!(self.core as c, self.messenger as m => move || {
+            for uri in uris {
+                if let Some(path) = uri.strip_prefix(FILE_SCHEME) {
+                    if let Err(err) = c.import_file(parent, path.to_string(), Some(Box::new(f.clone()))) {
+                        m.send_err_dialog("Import files", err);
+                        break;
+                    };
+                } else {
+                    panic!("impossible");
```

Comment:
instead of `panic!` can't we just `return`? You can `eprintln!` as well so we notice if this ever happens.